traveltimepy/requests/common.py

```diff
@@ -0,0 +1,376 @@
+    and determining how the off-road portions of journeys are calculated and reported.
+
+    Attributes:
+        penalty: Controls whether off-road walking distances are included in total journey metrics.
+                When enabled, includes "first and last mile" walking times/distances for realistic door-to-door calculations.
+        accept_roads: Defines which road types are valid as journey start/end points.
+                     Determines whether journeys can snap to vehicle-only roads or only pedestrian-accessible roads.
+    """
+
+    penalty: Optional[SnappingPenalty] = SnappingPenalty.ENABLED
+    accept_roads: Optional[SnappingAcceptRoads] = (
+        SnappingAcceptRoads.BOTH_DRIVABLE_AND_WALKABLE
+    )
+
+
+class ProtoProperty(int, Enum):
+    DISTANCE = 1
+
+
+class FullRange(BaseModel):
+    """
+    Configures time range search parameters for journey planning.
+
+    When enabled, allows searches to consider journeys within a specified
+    time window rather than at an exact time only, providing multiple journey options.
+    """
+
+    enabled: bool
+    """
+    Controls whether the time range search is active.
+
+    When enabled, adds a time window to the specified departure or arrival time.
+    Journey results will include any options that depart/arrive within this window.
+
+    Note:
+    - Disabled by default
+    - Only supported for public transportation modes (public_transport, coach, bus,
+      train, driving+train, driving+public_transport, cycling+public_transport)
+    - Ignored for other transportation modes
+    """
+
+    max_results: int = Field(gt=0, le=5)
+    """
+    Maximum number of results to return. Limited to 5 results.
+    Must be greater than 0.
+    """
+
+    width: int = Field(gt=0, le=43200)
+    """
+    Defines the width of the time range window in seconds.
+
+    Behavior varies based on whether searching by departure or arrival time:
+    - For departure time: Window extends forward (e.g., 9:00am with 1-hour width
+      includes journeys departing 9:00am-10:00am)
+    - For arrival time: Window extends backward (e.g., 9:00am with 1-hour width
+      includes journeys arriving 8:00am-9:00am)
+
+    Must be greater than 0. Maximum allowed value: 43,200 seconds (12 hours).
+    """
+
+
+class Range(BaseModel):
+    """
+    Configures time range parameters for route searching.
+
+    When enabled, allows searching for routes departing or arriving within a time window
+    rather than at a single specific time, returning a combined result that represents
+    all possible journeys within that window.
+
+    Time range functionality is primarily applicable to scheduled transportation modes:
+    public_transport, coach, bus, train, and driving+train combinations.
+    For other transportation modes (walking, cycling, driving), these parameters are ignored.
+    """
+
+    enabled: bool
+    """
```

Comment:
I think the documentation strings should be *before* the parameters, not after.

Reply:
If I move documentation string, then IDE doesn't find it. It would be possible to rework this to docustring though, but then probably documentation should be shorter.